docs/build.md

```diff
@@ -36,7 +36,7 @@ To enable ArborX examples, configure with
     -D ARBORX_ENABLE_EXAMPLES=ON
 ```
 
-To validate the ArborX build, install Boost version 1.68
+To validate the ArborX build, install Boost version 1.67 or 1.68
```

Comment:
Are we sure about the `1.67` requirement?  I can't remember where it comes from.
I can't tell what would break when I look in the [changelog](https://www.boost.org/users/history/version_1_67_0.html)

Reply:
I can't recall where it's coming from, but it is hardcoded in our tests.

Reply:
This is fine for now, as we know these 2 versions work in all configurations.